docs/rfcs/0002-ecommerce.md

```diff
@@ -0,0 +1,176 @@
+the [CyberSource payment processor backend](https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/payment/processors/cybersource.py#L369).
+This adds some complexity since there are significant differences
+between how they interact with the web application's frontend and backend.
+
+##### Recommendation
+
+Unless MIT will support other payment processors in the near future, we should
+design to use CyberSource similar to what we do in MicroMasters.
+
+#### Models
+
+MicroMasters only allows a user to purchase one course at a time so there
+is no concept of a cart or basket. In MicroMasters the `Order` object stores
+the information about the order and has a foreign key to `Line` to store
+information about multiple purchased items. However since the user can only purchase one course at
+once, there is only one `Line` per `Order` in MicroMasters.
+
+It's a requirement for mitxpro to support multiple items per order so we'll need a cart or basket
+for the user to keep track of their order. edX uses a `Basket` object for this purpose, which subclasses
+the `AbstractBasket` class from django-oscar. When a user clicks the button to verify a course, they
+are redirected to the `basket/add/?sku=SKU` URL which is handled by django-oscar to add the
+item to the basket.
+
+##### Recommendation
+
+Whether or not we want to use django-oscar, we will need models like these (using django-oscar names here):
+ - `Order` for an instance of a purchase
+ - `Basket` for a user's shopping cart. This should be connected to `Order`.
+ - `Line` for an item in an `Basket`
+ - `Product` to represent a purchasable item. In MicroMasters we skip this and 
+ instead set the course key in the `Line` directly but should have this object
+ to provide future flexibility and to let us decouple the price and the course information.
+ - `Voucher` to describe a coupon and how it can be used. A foreign key to `Basket` can be used
+ to indicate that a user wants to use the coupon for the purchase.
+ - `Receipt` to store CyberSource post-back information.
+ 
+And maybe something for purchase orders as well? We need some way to distinguish those from
+CyberSource purchases.
+
+Do we need audit tables? If so I think we can do this similar to how we did it for MicroMasters.
+
+#### django-oscar
+
+edX builds off of many of the abstract models that django-oscar provides, and they also subclass
+some of the django-oscar views. It makes sense to ask if we want to make use of these models, views or
+other functions rather than rolling our own like we did with MicroMasters.
+
+I think the big question is whether there is any functionality that could be useful and if using it saves times compared
+to rolling our own ecommerce infrastructure like we do on MicroMasters. While edX makes extensive use of django-oscar,
+they also have a lot of code which is written on top of django-oscar to handle their cases.
+
+##### Recommendation
+
+???
```

Comment:
Without having looked at the edX library yet, it sounds like the only argument for using their library is that they use `django-oscar` and added some functionality to it. Is there anything complicated in our use case that edX has solved in their library? If the answer is "no" or even "maybe", I would lean toward rolling our own. I would rather we solve for our use cases with `django-oscar` than start from edX's solution which probably solves a bunch of problems that are not relevant to this app.

Reply:
I agree, I don't think we should adapt edX's code for our use here.